ypathgen/pathgen.go

```diff
@@ -791,21 +893,41 @@ func generateDirectorySnippet(directory *ygen.Directory, directories map[string]
 	if len(errs) == 0 {
 		errs = nil
 	}
-	return GoPathStructCodeSnippet{
+	snippet := GoPathStructCodeSnippet{
 		PathStructName:    structData.TypeName,
 		StructBase:        structBuf.String(),
 		ChildConstructors: methodBuf.String(),
-	}, errs
+		Package:           goPackageName(directory, splitByModule, trimOCPkg, pkgName, fakeRootPkgName),
+	}
+	for dep := range deps {
+		snippet.Deps = append(snippet.Deps, dep)
+	}
+	snippets := []GoPathStructCodeSnippet{snippet}
+
+	for pkg, build := range listBuilderAPIBufs {
+		if code := build.String(); code != "" {
+			snippets = append(snippets, GoPathStructCodeSnippet{
+				PathStructName:    structData.TypeName,
+				Package:           pkg,
+				ChildConstructors: code,
+			})
+		}
+
+	}
+	return snippets, errs
 }
 
 // generateChildConstructors generates and writes to methodBuf the Go methods
-// that returns an instantiation of the child node's path struct object. It
-// takes as input the buffer to store the method, a directory, the field name
+// that returns an instantiation of the child node's path struct object.
+// When this is called on the fakeroot, the list Builder API methods need to be
+// in another package than the rest of the methods. In all other cases, methodBuf and
```

Comment:
the list builder API's methods need to be in the child package as opposed to the fakeroot's package.

Reply:
done